### PR TITLE
fix(skymp5-server): correctly handle IO error in FileDatabase upsert

### PR DESCRIPTION
**Pull request by nic11**

### CODE REVIEW

File: skymp5-server/cpp/server_guest_lib/FileDatabase.cpp [hunk 1]
```diff
@@ -21,17 +21,20 @@ FileDatabase::FileDatabase(std::string directory_,
...
   auto p = pImpl->changeFormsDirectory;
+  size_t nUpserted = 0;
 
   for (auto& changeForm : changeForms) {
     auto filePath = p / changeForm.formDesc.ToString('_');
     std::ofstream f(filePath);
     if (!f.is_open()) {
       pImpl->logger->error("Unable to open file {}", filePath.string());
+    } else {
+      f << MpChangeForm::ToJson(changeForm).dump();
```
[the remark this answers is not in the document]
Comment (Pospelove) [reply skymp5-server/cpp/server_guest_lib/FileDatabase.cpp:32]: @nic11 any progress on this?

[the remark this answers is not in the document]
Comment (nic11) [reply skymp5-server/cpp/server_guest_lib/FileDatabase.cpp:32]: Will do a bit later. Some kind of perfectionism prevents me from doing this like `if ... else { f << ...; if ... else ... }`

File: skymp5-server/cpp/server_guest_lib/FileDatabase.cpp [hunk 2]
```diff
@@ -21,17 +21,24 @@ FileDatabase::FileDatabase(std::string directory_,
 size_t FileDatabase::Upsert(const std::vector<MpChangeForm>& changeForms)
 {
   auto p = pImpl->changeFormsDirectory;
+  size_t nUpserted = 0;
 
   for (auto& changeForm : changeForms) {
     auto filePath = p / changeForm.formDesc.ToString('_');
     std::ofstream f(filePath);
+    if (f) {
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/FileDatabase.cpp:29]: failbit is set if error while opening occured, so this code should be correct

https://en.cppreference.com/w/cpp/io/basic_ofstream/open

**Comment by nic11** [comment]
Fixed the [issue above](https://github.com/skyrim-multiplayer/skymp/pull/89#discussion_r674443946). Didn't run tests locally though. Do `FileDatabase` tests run in CI?

File: skymp5-server/cpp/server_guest_lib/FileDatabase.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (nic11) [reply skymp5-server/cpp/server_guest_lib/FileDatabase.cpp]: fixed

**Comment by Pospelove** [comment]
It is being tested: https://github.com/skyrim-multiplayer/skymp/blob/3b92b19e903540bc773f8a04b39da9049e6b73d7/skymp5-server/cpp/unit/SaveStorageTest.h#L16

**Comment by Pospelove** [comment]
Looks good to me

**Pull request merged**